units/src/amount.rs

```diff
@@ -73,11 +64,8 @@ impl Denomination {
             Denomination::CentiBitcoin => -6,
             Denomination::MilliBitcoin => -5,
             Denomination::MicroBitcoin => -2,
-            Denomination::NanoBitcoin => 1,
-            Denomination::PicoBitcoin => 4,
             Denomination::Bit => -2,
             Denomination::Satoshi => 0,
-            Denomination::MilliSatoshi => 3,
```

Comment:
To better convey that we no longer support sub-satoshi units we should return `u8` from `precision`. I suspect it'll clean up some other code as well.